swiggyclone/service/views.py

```diff
@@ -11,67 +12,34 @@ class RestaurentViewSet(viewsets.ModelViewSet):
     serializer_class = RestaurentSerializer
 
+
 class BranchViewSet(viewsets.ModelViewSet):
     serializer_class = BranchSerializer
+
     def get_queryset(self):
         return Branch.objects.filter(restaurent_id=self.kwargs['restaurent_pk'])
 
+    def perform_create(self, serializer):
+        serializer.save(restaurent_id=self.kwargs['restaurent_pk'])
+
     @action(detail=True, methods=['POST'])
     def placeOrder(self, request, pk, restaurent_pk):
         data = request.data
         data["user"] = request.user
-        orderSerializers = OrderSerializers(data=data)
-        total_price = 0
-        orderValid = True
-        if orderSerializers.is_valid():
-            orderDiscriptions = data['orderDiscriptions']
-            orderedBranch = data['branch']
-            for orderDiscription in orderDiscriptions:
-                orderFoodItem = orderDiscription['foodItem']
-                orderQuantity = orderDiscription['quantity']
-                try: 
-                    instance = FoodItem.objects.filter(branch_id=orderedBranch, id=orderFoodItem)
-                except:
-                    orderValid = False
-                if orderValid:
-                    serialize = FoodItemSerializer(instance.first())
-                    foodItemQuantity = serialize.data["quantity"]
-                    if foodItemQuantity - orderQuantity < 0:
-                        orderValid = False
-                          
-            if orderValid:
-                for orderDiscription in orderDiscriptions:
-                    orderFoodItem = orderDiscription['foodItem']
-                    orderQuantity = orderDiscription['quantity']
-                    instance = FoodItem.objects.filter(branch_id=orderedBranch, id=orderFoodItem)
-                    foodItemPrice = instance.first().price
-                    finalQuantity = instance.first().quantity - orderQuantity
-                    update_serializer = FoodItemSerializer(instance.first(), data={"quantity": finalQuantity},
-                                                           partial=True)
-                    if update_serializer.is_valid():
-                        update_serializer.save()
-                        total_price = total_price + orderQuantity*foodItemPrice
-                        
-                    else:
-                        return Response(update_serializer.errors, status=400)
-                data["user"] = request.user
-                data["total_price"] = total_price
-                finalOrderSerializer = OrderSerializers(data=data)
-                if finalOrderSerializer.is_valid():   
-                    finalOrderSerializer.save()
-                else:
-                    orderValid = False
-                    return Response(orderSerializers.errors, status=404)
+        data['branch'] = self.kwargs['pk']
+        serializer = OrderSerializers(data=data)
+        if serializer.is_valid():
```

Comment:
serializer.is_valid(raise_exception=True)
serializer.save()

No need to write else condition